pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -131,14 +131,31 @@ func (esc *endpointSliceController) onEventInternal(_, ep *v1.EndpointSlice, eve
 			continue
 		}
 
-		esc.c.opts.XDSUpdater.ConfigUpdate(&model.PushRequest{
-			Full: features.EnableHeadlessService,
-			// TODO: extend and set service instance type, so no need to re-init push context
-			ConfigsUpdated: sets.New(model.ConfigKey{Kind: kind.ServiceEntry, Name: modelSvc.Hostname.String(), Namespace: svc.Namespace}),
-
-			Reason: model.NewReasonStats(model.HeadlessEndpointUpdate),
-		})
-		break
+		// pure HTTP headless services should not need a full push since they do not
+		// require a Listener based on IP: https://github.com/istio/istio/issues/48207
+		pureHttp := false
+		for _, p := range modelSvc.Ports {
+			if p.Protocol.IsHTTP() {
+				pureHttp = true
+			} else {
+				pureHttp = false
+				break
```

Comment:
nit: 
init pureHttp = true

if !p.Protocol.IsHTTP(){
 pureHttp = false
}

Reply:
yeah, I was being overly cautious here - but I suppose model.Service should have a protocol properly detected here that `IsXXXX` functions should not skip any edge cases